pkg/fadvis/manager.go

```diff
@@ -0,0 +1,186 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package fadvis manages file access strategies based on runtime memory thresholds.
+package fadvis
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/dustin/go-humanize"
+
+	"github.com/apache/skywalking-banyandb/banyand/protector"
+	"github.com/apache/skywalking-banyandb/pkg/fs"
+	"github.com/apache/skywalking-banyandb/pkg/logger"
+	"github.com/apache/skywalking-banyandb/pkg/run"
+)
+
+// ThresholdProvider is an interface for providing file size thresholds.
+type ThresholdProvider interface {
+	// GetThreshold returns the file size threshold.
+	GetThreshold() int64
+}
+
+// Global FadvisManager instance.
+var defaultManager *Manager
+
+// SetManager sets the global FadvisManager instance.
+func SetManager(m *Manager) {
+	defaultManager = m
+}
+
+// GetManager returns the global FadvisManager instance.
+func GetManager() *Manager {
+	return defaultManager
+}
+
+// Manager manages the fadvis threshold and periodically updates it.
+type Manager struct {
```

Comment:
Your "Manager" seems "over-designed." The function `protector.GetThreshold` is lightweight, allowing you to use it directly in block_writer instead of storing the value in this Manager as a vault.